src/Estimators/src/AccelerometerMeasurementDynamics.cpp

```diff
@@ -0,0 +1,224 @@
+
+    m_kinDynWrapperList = kinDynWrapperList;
+
+    m_isSubModelListSet = true;
+
+    return true;
+}
+
+bool RDE::AccelerometerMeasurementDynamics::checkStateVariableHandler()
+{
+    constexpr auto errorPrefix = "[AccelerometerMeasurementDynamics::checkStateVariableHandler]";
+
+    if (m_useBias)
+    {
+        if (!m_stateVariableHandler.getVariable(m_biasVariableName).isValid())
+        {
+            log()->error("{} The variable handler does not contain the expected state with name `{}`.", errorPrefix, m_biasVariableName);
+            return false;
+        }
+    }
+
+    return true;
+}
+
+bool RDE::AccelerometerMeasurementDynamics::update()
+{
+    // compute joint acceleration per each sub-model containing the accelerometer
+    for (int subDynamicsIdx = 0; subDynamicsIdx < m_subModelList.size(); subDynamicsIdx++)
+    {
+        if (m_subModelList[subDynamicsIdx].getModel().getNrOfDOFs() > 0)
+        {
+            for (int jointIdx = 0; jointIdx < m_subModelList[subDynamicsIdx].getJointMapping().size(); jointIdx++)
+            {
+                m_subModelJointAcc[subDynamicsIdx][jointIdx] = m_ukfInput.robotJointAccelerations[m_subModelList[subDynamicsIdx].getJointMapping()[jointIdx]];
+            }
+        }
+    }
+
+    // J_dot nu + base_J v_dot_base + joint_J s_dotdot - acc_Rot_world gravity + bias
+    for (int index = 0; index < m_subModelsWithAccelerometer.size(); index++)
+    {
+        m_JdotNu = m_kinDynWrapperList[m_subModelsWithAccelerometer[index]]->getAccelerometerBiasAcceleration(m_name);
+
+        m_JvdotBase = m_kinDynWrapperList[m_subModelsWithAccelerometer[index]]->getAccelerometerJacobian(m_name).block(0, 0, 6, 6) *
+                m_kinDynWrapperList[m_subModelsWithAccelerometer[index]]->getBaseAcceleration().coeffs();
+
+        m_accRg = m_kinDynWrapperList[m_subModelsWithAccelerometer[index]]->getAccelerometerRotation(m_name).rotation() * m_gravity;
+
+        m_linVel = m_kinDynWrapperList[m_subModelsWithAccelerometer[index]]->getAccelerometerVelocity(m_name).coeffs().segment(0, 3);
+        m_angVel = m_kinDynWrapperList[m_subModelsWithAccelerometer[index]]->getAccelerometerVelocity(m_name).coeffs().segment(3, 3);
+
+        m_vCrossW = m_linVel.cross(m_angVel);
+
+        m_updatedVariable.segment(index * m_covSingleVar.size(), m_covSingleVar.size()) = m_JdotNu.segment(0, 3) + m_JvdotBase.segment(0, 3) - m_vCrossW - m_accRg;
+
+        if (m_useBias)
+        {
+            m_updatedVariable.segment(index * m_covSingleVar.size(), m_covSingleVar.size()) = m_updatedVariable.segment(index * m_covSingleVar.size(), m_covSingleVar.size()) + m_bias;
+        }
+
+        if (m_subModelList[m_subModelsWithAccelerometer[index]].getJointMapping().size() > 0)
+        {
+            m_Jsdotdot = m_kinDynWrapperList[m_subModelsWithAccelerometer[index]]->getAccelerometerJacobian(m_name).
+                    block(0, 6, 6, m_subModelJointAcc[m_subModelsWithAccelerometer[index]].size()) *
```

Comment:
Also here you may use a smarter block operation: https://eigen.tuxfamily.org/dox/group__TutorialBlockOperations.html